tests/main.cpp

```diff
@@ -57,14 +58,21 @@ TEST_CASE("Simulate & Send Transaction") {
 
 TEST_CASE("Request Airdrop") {
   const solana::Keypair keyPair = solana::Keypair::fromFile(KEY_PAIR_FILE);
-  auto connection = solana::rpc::Connection(solana::DEVNET);
-
-  auto prev_sol = connection.getBalance(keyPair.publicKey);
-  auto result = connection.requestAirdrop(keyPair.publicKey, 1000000000);
-  std::this_thread::sleep_for(std::chrono::seconds(15));
-  auto new_sol = connection.getBalance(keyPair.publicKey);
-  // TODO: validate using confirmTransaction
-  CHECK_GT(new_sol["value"], prev_sol["value"]);
+  const auto connection = solana::rpc::Connection(solana::DEVNET);
+  // request Airdrop
+  const auto prev_sol = connection.getBalance(keyPair.publicKey);
+  const auto signature = connection.requestAirdrop(keyPair.publicKey, 50001);
+  // check signature status
+  while (true) {
+    const auto res = connection.getSignatureStatus(signature, true).value;
```

Comment:
this will cause infinite loop in case if it really fails. And CI will never end.

Reply:
We were thinking of this too. What should be the best way to solve this ?

Reply:
I have added a timeout for 15 secs for the while true loop. Should be fixed now

Reply:
waiting for #44